### PR TITLE
Show all available annotation topics in the annotations dropdown

### PR DESCRIPTION
**Pull request by defunctzombie**
**User-Facing Changes**
Users can select any valid annotation topic using the annotation dropdown. The relevant annotation topics are still first.

Screen Shot 2022-02-18 at 1 52 04 PM

**Description**
This change updates the Image panel annotations dropdown to show all available annotation topics. The "related" topics appear first. Those are the ones that are under the same "camera namespace" (i.e same topic prefix). Then the remaining topics after.
    
This change also switches the Dropdown component to the MUI Select component for the dropdown.
    
Fixes: #1675

### CODE REVIEW

**Comment by amacneil** [comment]
Should we hide the other topics under  "more" link or something? Do you think people will find it confusing seeing both related and unrelated annotation topics?

**Comment by defunctzombie** [comment]
> Do you think people will find it confusing seeing both related and unrelated annotation topics?

Unrelated is somewhat of a "ros-ism" that you organize the topics in this way. But maybe enough folks adopt this convention that it works out. They are "unrelated" only in the sense that in ros you namespace the topics like this because the tooling does it by convention. Nothing "really" stops you from making topics like `/annotations/camera_front` that mirror `/camera_front/image` but I agree its somewhat contrived and weird.

We could make this a "more" sub-menu. We need to settle on what type of component this is tho for that. Right now its a "Select" component - that does not have "submenus" I think. We can make it have them tho.

**Comment by defunctzombie** [comment]
@jtbandes @jhurliman for additional thoughts and usecases

**Comment by amacneil** [comment]
Agreed - hard to say what is right/wrong.

Maybe - IF there are matching sub-topics we show those with the rest under a "more" link, and if there are none we show all topics by default?

**Comment by jtbandes** [comment]
I think we have a lot of cleanup to do on the styling of the new dropdown, but the interaction looks ok to me. In the future I imagine we might want to make this look/work more like the 3D panel topic picker since it is kind of doing the same thing.

**Comment by amacneil** [comment]
image

**Comment by defunctzombie** [comment]
@2metres what's the status of this PR? I see that we've removed the disablePortal stuff. Is there some plan for dealing with the floating toolbar? It seems this PR is stuck in some limbo and we should figure out what we need to push it over the finish line.

**Comment by 2metres** [comment]
Talked to @foxymiles about it and he has some thoughts about how to implement the less glitchy hover and menu states. But wanted to do it in a separate PR as a fast follow.

**Comment by defunctzombie** [comment]
Plan is to merge this PR without the disablePortal and figure out an approach for resolving the annotations dropdown glitch. This resolves the image topic glitch since the menu closes after selecting the topic.

File: packages/studio-base/src/components/PanelToolbar/index.tsx [hunk 1]
```diff
@@ -58,11 +58,7 @@ const useStyles = makeStyles((theme) => ({
       width: "100%",
       zIndex: 5000,
       backgroundColor: "transparent",
-      pointerEvents: "none",
```
Comment (2metres) [inline packages/studio-base/src/components/PanelToolbar/index.tsx:61]: I think removing this could cause problems with clicking the underlying canvas

**Review by 2metres — APPROVED** [review]
Just one minor css thing but looks good to me

File: packages/studio-base/src/components/PanelToolbar/index.tsx (hunk 1 above)
Comment (defunctzombie) [reply packages/studio-base/src/components/PanelToolbar/index.tsx:61]: I am still able to click on objects in the canvas. Are you not?

Comment (defunctzombie) [reply packages/studio-base/src/components/PanelToolbar/index.tsx:61]: Indeed it did cause problems. https://github.com/foxglove/studio/pull/2989

**Pull request merged**